### PR TITLE
macOS: install GNU tar

### PR DESCRIPTION
**Pull request by jonas-schievink**
CI runs on macOS have started to fail due to https://github.com/actions/cache/issues/403. Use GNU tar instead of BSD tar to work around that.

### CODE REVIEW

File: bors.toml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 status = [
     "Rust (ubuntu-latest)",
     "Rust (windows-latest)",
-    # "Rust (macos-latest)",
```
Comment (matklad) [inline bors.toml:4]: lol

**Comment by jonas-schievink** [comment]
bors retry

File: Cargo.lock [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
+# asdasdasdasd
```
Comment (lnicola) [inline Cargo.lock:3]: lol

**Pull request merged**